src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.cpp

```diff
@@ -1518,60 +1519,110 @@ bool ReconstructionEngine_sequentialSfM::checkAngles(const Vec3 &pt3D, const std
+//    std::advance(iter, i);
+//    const IndexT indexAll = *iter;
+    
+//    for(IndexT indexNew: newReconstructedViews)
+//    {
+//      if(indexAll == indexNew)
+//        continue;
+      
+//      const std::size_t I = std::min((IndexT)indexNew, indexAll);
+//      const std::size_t J = std::max((IndexT)indexNew, indexAll);
+//      // Find track correspondences between I and J
+//      const std::set<std::size_t> set_viewIndex = { I, J };
+//      track::TracksMap map_tracksCommonIJ;
+//      track::TracksUtilsMap::GetCommonTracksInImagesFast(set_viewIndex, _map_tracks, _map_tracksPerView, map_tracksCommonIJ);
+      
+//      if (map_tracksCommonIJ.empty())
+//        continue;
+        
+//      // Collect tracksIds
+//      std::set<std::size_t> commonTracksId;
+//      std::transform(map_tracksCommonIJ.begin(), map_tracksCommonIJ.end(),
+//                     std::inserter(commonTracksId, commonTracksId.begin()),
+//                     stl::RetrieveKey());
+      
+//#pragma omp critical        
+//      for (std::size_t trackId : commonTracksId)
+//      {
+//        mapTracksToTriangulate[trackId].insert(I);      
+//        mapTracksToTriangulate[trackId].insert(J);      
+//      } // tracks to reconstruct/update
+//    } // new reconstructed view
+//  } // all recontrusted views
+//}
+
+void ReconstructionEngine_sequentialSfM::getTracksToTriangulate(const std::set<IndexT>& previousReconstructedViews, 
+                                                                const std::set<IndexT>& newReconstructedViews, 
+                                                                std::map<IndexT, std::set<IndexT>> & mapTracksToTriangulate)
 {
-  ALICEVISION_LOG_DEBUG("Triangulating (mode: multi-view LO-RANSAC)... ");
+  std::set<IndexT> allReconstructedViews;
+  allReconstructedViews.insert(previousReconstructedViews.begin(), previousReconstructedViews.end());
+  allReconstructedViews.insert(newReconstructedViews.begin(), newReconstructedViews.end());
   
-  // -- Identify the track to triangulate :
-  // This map contains all the tracks that will be triangulated (for the first time, or not)
-  // These tracks are seen by at least one new reconstructed view.
-  std::map<IndexT, std::set<IndexT>> mapTracksToTriangulate; // <trackId, observations> 
-  std::vector<IndexT> setTracksId; // <trackId>
-  {
-    std::set<IndexT> allReconstructedViews;
-    allReconstructedViews.insert(previousReconstructedViews.begin(), previousReconstructedViews.end());
-    allReconstructedViews.insert(newReconstructedViews.begin(), newReconstructedViews.end());
+  std::set<IndexT> allTracksInNewViews;
+  track::TracksUtilsMap::GetTracksInImagesFast(newReconstructedViews, _map_tracksPerView, allTracksInNewViews);
   
-#pragma omp parallel for 
-    for (ptrdiff_t i = 0; i < static_cast<ptrdiff_t>(allReconstructedViews.size()); ++i)
+  std::set<IndexT>::iterator it;
+#pragma omp parallel private(it)
```

Comment:
It's a matter of taste but I found the previous way of using omp (based on indexes instead of iterators) easier to read.

Reply:
Yes it's easier to read, and this is what I used first, but _std::advance()_ consumed a lot of time. 

```cpp
#pragma omp parallel for schedule(dynamic)
  for (ptrdiff_t i = 0; i < static_cast<ptrdiff_t>(allTracksInNewViews.size()); ++i)
  {
    std::set<IndexT>::const_iterator iter = allTracksInNewViews.begin();
    std::advance(it, i);
    const std::size_t trackId = *it;
    // ...
}
```
With this method, the function needs more time than the one we want to replace.
Here is my motivations to use this method... :)

Reply:
Yes it's easier to read, and this is what I used first, but _std::advance()_ consumed a lot of time. 

```cpp
#pragma omp parallel for schedule(dynamic)
  for (ptrdiff_t i = 0; i < static_cast<ptrdiff_t>(allTracksInNewViews.size()); ++i)
  {
    std::set<IndexT>::const_iterator it = allTracksInNewViews.begin();
    std::advance(it, i);
    const std::size_t trackId = *it;
    // ...
}
```
With this method, the function needs more time than the one we want to replace.
Here is my motivations to use this method... :)

Reply:
Ah ok, I was not aware of a performance difference between the 2 approaches.